File: JavaScript-Histogram/Assignment.js
class MyString {

  constructor(str) {
    if (str == undefined) {
      str = "Hello, World!";
    }

    let re = /[W_]/g;
     this.str = str.toUpperCase().replace(re,'');
     this.letters = {};
  }

  frequency() {
    for (let i = 65; i <= 90; i++) {
      this.letters[String.fromCharCode(i)] = 0;
    }

    for (let i in this.str) {
      let count = this.str[i];
      this.letters[count]++;
    }
  }

  histogram() {
    for (let k in this.letters) {
      let bR = this.letters[k];
      let result = k + ": ";
      for (let i = 0; i < bR; i++) {
        result += "*";
      }

      console.log(result);
    }
  }
}

File: Ruby-Histogram/Assignment.rb
class MyString

  attr_accessor :str
  attr_reader :letters

  def initialize
    @str = 'Hello, World!'

  end

     def frequency
       @letters = @str.each_char.with_object(Hash.new(0)) { |hsh, key| hsh[key] += 1; hsh}
     end

        def histogram
          frequency()
          ("a".."z").each do |letter|
            i=0
            lineDown=""
            while i<@letters[letter]
              lineDown="#{lineDown}*"
              i = i + 1
            end
            puts "#{letter}: #{lineDown}"
        end
end
  mstr = MyString.new("Look out");
  mstr.histogram()
